### PR TITLE
Feat: Dev App

### PR DESCRIPTION
**Pull request by migsoffshorly**
Dev app for dynamically adding new files to the table of contents as well as dev tools to lint markdown content

changes:
- move md folders to content,
-  add chokidar to listen to changes, 
- add typescript, 
- add markdownlint, 
- add handling for updating the README.md, 
- create build step for typescript, 
- add vscode extension installation

### CODE REVIEW

File: scripts/lint.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import chalk from 'chalk'
+import { generateVSCodeLink, hasAnyArrayWithValues } from './helpers.js'
+import lintContent from './lintContent.js'
+
+async function lintMdFiles(path: string) {
+  const result = await lintContent([path])
+  const keys = Object.keys(result)
+  const values = Object.values(result)
+
+  if (hasAnyArrayWithValues(values)) {
```
Comment (jasonoffshorlydev) [inline scripts/lint.ts]: Maybe we can flatten the nested if statements here with inversion:
```
if (!hasAnyArrayWithValues(values)) return true
for (let i = 0; i < keys.length; i++) {
  ...
}
```

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import chalk from 'chalk'
+import chokidar from 'chokidar'
+import figlet from 'figlet'
+import { readFile, writeFile } from 'fs/promises'
+import getMdPaths from './scripts/getMdPaths.js'
+import { installVsCodeExtension } from './scripts/helpers.js'
+import lintMdFiles from './scripts/lint.js'
+import mapContent from './scripts/mapContent.js'
+;(async () => {
```
Comment (jasonoffshorlydev) [inline index.ts:9]: is the `;` here intentional?

File: scripts/lint.ts (hunk 1 above)
Comment (migsoffshorly) [reply scripts/lint.ts]: for sure. will update!

File: index.ts (hunk 2 above)
Comment (migsoffshorly) [reply index.ts:9]: It's for safety. Read it here: https://stackoverflow.com/questions/7365172/semicolon-before-self-invoking-function#:~:text=Putting%20a%20semicolon%20before%20the,become%20confused%20with%20function%20application.

It is also auto-inserted by eslint.

**Pull request merged**